### PR TITLE
request: cleanup request completion processing

### PR DESCRIPTION
**Pull request by danghvu**
This PR are possible cleanup that I found while working with the requests. They are mostly code refactoring to reduce code duplication and dead code.

### CODE REVIEW

**Comment by danghvu** [comment]
test:jenkins/ch3/tcp

**Comment by danghvu** [comment]
test:jenkins/ch3/tcp
test:jenkins/ch4/ucx

**Comment by danghvu** [comment]
@raffenet please review this when you got a chance

File: src/mpi/request/mpir_request.c [hunk 1]
```diff
@@ -127,12 +123,7 @@ int MPIR_Request_completion_processing(MPIR_Request * request_ptr, MPI_Status *
...
             {
-                int rc;
-
-                rc = MPIR_Grequest_query(request_ptr);
-                if (mpi_errno == MPI_SUCCESS) {
-                    mpi_errno = rc;
-                }
+                mpi_errno = MPIR_Grequest_query(request_ptr);
                 MPIR_Request_extract_status(request_ptr, status);
                 rc = MPIR_Grequest_free(request_ptr);
```
Comment (raffenet) [inline src/mpi/request/mpir_request.c]: @danghvu sorry for the long delay in reviewing. My only comment is here. I think we should do this same simplification for `MPIR_Grequest_free`. It and `MPIR_Grequest_query` return MPI error codes, so there is no need for the intermediate `rc` variable.

Comment (raffenet) [reply src/mpi/request/mpir_request.c]: OK I see what you mean now. Sorry for the long delay again. I'm going to rebase/retest and see if hopefully this is ready for merge.

**Comment by raffenet** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by raffenet** [comment]
This got out of date because of the workq PR. Rebased now. Re-testing...
test:jenkins/ch3/most
test:jenkins/ch4/most

**Pull request merged**